From 0b69a5e66e97e3e887bda5e6bf18e2c16da7c88f Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Mon, 27 May 2019 13:50:15 +0800
Subject: [PATCH] dbus: fix avc denials for system_dbusd_var_run_t

Patches from Fedora.

Fix:
type=AVC msg=audit(1558926483.188:35): avc:  denied  { read } for
pid=160 comm="systemd-network" name="dbus" dev="tmpfs" ino=13018
scontext=system_u:system_r:systemd_networkd_t
tcontext=system_u:object_r:system_dbusd_var_run_t tclass=dir
permissive=1

type=AVC msg=audit(1558926483.190:36): avc:  denied  { read } for
pid=176 comm="systemd-resolve" name="dbus" dev="tmpfs" ino=13018
scontext=system_u:system_r:systemd_resolved_t
tcontext=system_u:object_r:system_dbusd_var_run_t tclass=dir
permissive=1

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/services/dbus.if  | 59 ++++++++++++++++++++++++++++++++++++++++
 policy/modules/system/systemd.te |  4 +++
 2 files changed, 63 insertions(+)

diff --git a/policy/modules/services/dbus.if b/policy/modules/services/dbus.if
index c815791..8e118f2 100644
--- a/policy/modules/services/dbus.if
+++ b/policy/modules/services/dbus.if
@@ -593,3 +593,62 @@ interface(`dbus_unconfined',`
 
 	typeattribute $1 dbusd_unconfined;
 ')
+
+########################################
+## <summary>
+##	Delete all dbus pid files
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`dbus_delete_pid_files',`
+	gen_require(`
+		type system_dbusd_var_run_t;
+	')
+
+	files_search_pids($1)
+	delete_files_pattern($1, system_dbusd_var_run_t, system_dbusd_var_run_t)
+')
+
+########################################
+## <summary>
+##	Read all dbus pid files
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`dbus_read_pid_files',`
+	gen_require(`
+		type system_dbusd_var_run_t;
+	')
+
+	files_search_pids($1)
+	list_dirs_pattern($1, system_dbusd_var_run_t, system_dbusd_var_run_t)
+	read_files_pattern($1, system_dbusd_var_run_t, system_dbusd_var_run_t)
+')
+
+########################################
+## <summary>
+##	Read all dbus pid files
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`dbus_read_pid_sock_files',`
+	gen_require(`
+		type system_dbusd_var_run_t;
+	')
+
+	files_search_pids($1)
+	list_dirs_pattern($1, system_dbusd_var_run_t, system_dbusd_var_run_t)
+	read_sock_files_pattern($1, system_dbusd_var_run_t, system_dbusd_var_run_t)
+')
diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index 33cdede..7611450 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -759,6 +759,8 @@ systemd_log_parse_environment(systemd_networkd_t)
 optional_policy(`
 	dbus_system_bus_client(systemd_networkd_t)
 	dbus_connect_system_bus(systemd_networkd_t)
+	dbus_read_pid_files(systemd_networkd_t)
+	dbus_read_pid_sock_files(systemd_networkd_t)
 ')
 
 optional_policy(`
@@ -1067,6 +1069,8 @@ init_write_pid_socket(systemd_resolved_t)
 optional_policy(`
 	dbus_connect_system_bus(systemd_resolved_t)
 	dbus_system_bus_client(systemd_resolved_t)
+	dbus_read_pid_files(systemd_resolved_t)
+	dbus_read_pid_sock_files(systemd_resolved_t)
 ')
 
 #########################################
-- 
2.7.4

